wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -240,7 +245,7 @@ newtype AssertionError = AssertionError T.Text
 data Event n a where
     -- | An direct action performed by a wallet. Usually represents a "user action", as it is
     -- triggered externally.
-    WalletAction :: Wallet -> n () -> Event n [Tx]
+    WalletAction :: Wallet -> n a -> Event n [Tx]
```

Comment:
I feel like I said this before, but I want this to be `()` because we don't use the result value, so I want to force people to explicitly discard it.

Reply:
I'm not sure this is possible because we need to notify all wallets of these transactions don't we?

Reply:
I'm talking about the parameter you changed from `()` to `a`, not the `[Tx]` at the end, which yes we do indeed need.